Add New Metric to Show Number of Running PipelineRuns

zakisk opened the pull request:
added new gauge metric to show number of running pipelineruns. added tests and docs accordingly.

https://issues.redhat.com/browse/SRVKP-6375

![running pipelineruns](https://github.com/user-attachments/assets/8fe6682d-0459-44c3-b55d-4c1b2940088e)


# Changes <!-- 🎉🎉🎉 Thank you for the PR!!! 🎉🎉🎉 -->

<!-- Describe your changes here- ideally you can get that description straight from
your descriptive commit message(s)! -->

# Submitter Checklist

- [ ] 📝 Please ensure your commit message is clear and informative. For guidance on crafting effective commit messages, refer to the How to write a git commit message guide. We prefer the commit message to be included in the PR body itself rather than a link to an external website (ie: Jira ticket).

- [ ] ♽ Before submitting a PR, run make test lint to avoid unnecessary CI processing. For an even more efficient workflow, consider installing [pre-commit](https://pre-commit.com/) and running pre-commit install in the root of this repository.

- [ ] ✨ We use linters to maintain clean and consistent code. Please ensure you've run make lint before submitting a PR. Some linters offer a --fix mode, which can be executed with the command make fix-linters (ensure [markdownlint](https://github.com/DavidAnson/markdownlint) and [golangci-lint](https://github.com/golangci/golangci-lint) tools are installed first).

- [ ] 📖 If you're introducing a user-facing feature or changing existing behavior, please ensure it's properly documented.

- [ ] 🧪 While 100% coverage isn't a requirement, we encourage unit tests for any code changes where possible.

- [ ] 🎁 If feasible, please check if an end-to-end test can be added. See [README](https://github.com/openshift-pipelines/pipelines-as-code/blob/main/test/README.md) for more details.

- [ ] 🔎 If there's any flakiness in the CI tests, don't necessarily ignore it. It's better to address the issue before merging, or provide a valid reason to bypass it if fixing isn't possible (e.g., token rate limitations).

chmouel commented on pkg/metrics/metrics.go:
```diff
@@ -19,6 +26,10 @@ var prDurationCount = stats.Float64("pipelines_as_code_pipelinerun_duration_seco
 	stats.UnitDimensionless)
 
+var runningPRCount = stats.Float64("pipelines_as_code_running_pipelineruns_count",
+	"number of running pipeline runs by pipelines as code",
```
```suggestion
	"number of running  pipelineruns runs by pipelines as code",
```

chmouel commented on pkg/metrics/metrics.go:
```diff
@@ -31,79 +42,98 @@ type Recorder struct {
-		ReportingPeriod: 30 * time.Second,
-	}
+			// Default to 10s intervals.
+			ReportingPeriod: 10 * time.Second,
```
how did you choose this number?

zakisk replied:
@chmouel oh, that is mistakenly.

zakisk replied:
Actually I was running some pipelineruns to get metrics which were finishing in 20 seconds and original interval period was 30 seconds so injection code wasn't running for last pipelineruns and metric was having 1 only (it was being updated to 0 after all PR has done) that's why I changed it to 10 seconds, but I don't think that in production there would be any PipelineRun which will ends in 20 seconds.

zakisk committed:
Add new metric to show number of running pipelineruns

added new gauge metric to show number of running pipelineruns.
added tests and docs accordingly.

https://issues.redhat.com/browse/SRVKP-6375

Signed-off-by: Zaki Shaikh <zashaikh@redhat.com>

zakisk commented:
/test

zakisk commented:
/test
